rfcs/0140-simple-package-paths.md

```diff
@@ -0,0 +1,363 @@
+- It's not possible anymore to be a [GitHub code owner](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) of category directories.
+- The existing categorization of packages gets lost. Counter-arguments:
+  - It was never that useful to begin with.
+    - The categorization was always incomplete, because packages defined in the language package sets often don't get their own categorized file path.
+    - It was an inconvenient user interface, requiring a checkout or browsing through GitHub
+    - Many packages fit multiple categories, leading to multiple locations to search through instead of one
+  - There's other better ways of discovering similar packages, e.g. [Repology](https://repology.org/)
+- This breaks `builtins.unsafeGetAttrPos "hello" pkgs`. Counter-arguments:
+  - We have to draw a line as to what constitutes the public interface of Nixpkgs. We have decided that making attribute position information part of that is not productive. For context, this information is already accepted to be unreliable at the language level, noting the `unsafe` part of the name.
+  - Support for this could be added to Nix (make `builtins.readDir` propagate file as a position)
+
+# Alternatives
+[alternatives]: #alternatives
+
+## A different unit base directory
+
+Context: `pkgs/unit` is the unit base directory
+
+Alternatives:
+- Use `unit` in the root directory instead
+  - (+) This is future proof in case we want to make the directory structure more general purpose
+    - (-) We don't yet know if we want that, so this is out of scope for now
+- Use `pkgs` instead, so that the `${shard}`'s are siblings to the other current directories in `pkgs` such as `top-level`, with the intention that the other directories would be hopefully removed at some point, then only leaving the shards in `pkgs`
+  - (+) If we remove the other directories at some point, only the `${shard}`'s will be left in `pkgs`
+  - (-) This leads to ambiguities between the directories from the standard and the other directories, requiring special handling in the code and CI, leading to complexities.
+  - (-) This would be harder to document and explain to people, since one always has to disregard all non-unit directories, with no obvious justification
+  - (-) Currently we cannot apply this standard to all definitions in `pkgs`, in particular nested packages like `pythonPackages.*`, non-`callPackage`'d definitions like `copyDesktopItems` and non-derivations like `fetchFromGitHub`.
+    Depending on how we want to handle those, it might make more sense to keep `pkgs/unit` or to use `pkgs` directly once all legacy paths are migrated away to another top-level directory, we don't yet know. `pkgs/unit` will be easier to migrate to `pkgs` than the other way around though.
+- Another name than `unit`, proposed were `auto`, `pkg`, `mod`, `component`, `part`, `comp`, `app`, `simple`
```

Comment:
```suggestion
- Another name than `unit`, proposed were `auto`, `pkg`, `mod`, `component`, `part`, `comp`, `app`, `simple`, `shard`
```

Reply:
In light of the current paint-slinging: what about `pkgs/by-name`?

Reply:
@infinisil It would help noting the reasons for each why it's not the preferred choice.

@rhendric `by-name` this is really good, it tells me what it's supposed to mean.